src/algorithms/PVT/gnuradio_blocks/rtklib_pvt_gs.cc

```diff
@@ -93,17 +93,20 @@ namespace bc = boost::integer;
 
 rtklib_pvt_gs_sptr rtklib_make_pvt_gs(uint32_t nchannels,
     const Pvt_Conf& conf_,
-    const rtk_t& rtk)
+    const rtk_t& rtk,
+    SpoofingDetector spoofing_detector)
 {
     return rtklib_pvt_gs_sptr(new rtklib_pvt_gs(nchannels,
         conf_,
-        rtk));
+        rtk,
+        spoofing_detector));
 }
 
 
 rtklib_pvt_gs::rtklib_pvt_gs(uint32_t nchannels,
     const Pvt_Conf& conf_,
-    const rtk_t& rtk) : gr::sync_block("rtklib_pvt_gs",
+    const rtk_t& rtk,
+    SpoofingDetector spoofing_detector) : gr::sync_block("rtklib_pvt_gs",
```

Comment:
and here? is the pass by value intentional?

You end up with 4 copies of this object created. First you allocated one on the heap to pass to the factory method (it was leaked). Then you made a copy of that by passing into the factory by value. Here you make a 3rd copy, by passing by value here. Finally, you make a 4th copy below. 

All these copies seem a bit pointless.